platform/kube/register.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	// For most common ports allow the protocol to be guessed, this isn't meant
+	// to replace /etc/services. Fully qualified proto[-extra]:port is the
+	// recommended usage.
+	portsToName = map[int32]string{
+		80:   "http",
```

Comment:
well, the problem is guessing http2 and grpc... I think explicit requirement to specify port name would be better, but 
I don't mind a bit of magic...

Reply:
this is just a convenience, you can specify protocol if you wish (as commented above)